ambry-replication/src/test/java/com.github.ambry.replication/ReplicationTest.java

```diff
@@ -807,6 +819,66 @@ public void replicaThreadTest() throws Exception {
     // 1 expired + 1 corrupt + 1 put (never present) + 1 deleted (never present) expected missing buffers
     verifyNoMoreMissingKeysAndExpectedMissingBufferCount(remoteHost, localHost, replicaThread, replicasToReplicate,
         idsToBeIgnoredByPartition, storeKeyConverter, expectedIndex, expectedIndex + 1, 4);
+
+    // tests to verify replica thread throttling and idling functions in the following steps:
+    // 1. all replicas are in sync, thread level sleep and replica quarantine are both enabled.
+    // 2. add put messages to some replica and verify that replication for replicas remain disabled.
+    // 3. forward the time so replication for replicas are re-enabled and check replication resumes.
+    // 4. add more put messages to ensure replication happens continuously when needed and is throttled appropriately.
+
+    // 1. verify that the replica thread sleeps and replicas are temporarily disable when all replicas are synced.
+    List<List<RemoteReplicaInfo>> replicasToReplicateList = new ArrayList<>(replicasToReplicate.values());
+    replicaThread.replicate(replicasToReplicateList);
+    // replicate is called twice to trigger the sleep (replication for all replica is disabled then replicate is put to sleep).
```

Comment:
minor: you could check the `RemoteReplicaInfo` and make sure the correct reenable time has been set.